### PR TITLE
test: split and improve ObjectFieldMapTypeHandlerFactoryTests

### PR DESCRIPTION
**Pull request by jdrueckert**
- split into multiple tests to avoid testing different aspects in single test
- remove usage of implicit member type knowledge in assertion phase
- add negative assertions to ensure only expected calls are made
- improve test case documentation

Once #5154 is merged, we'll need to update the suppression of `UUF_UNUSED_FIELD` spotbugs findings for `SingleTypeClass` and `MultiTypeClass`.

### CODE REVIEW

File: subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/coreTypes/factories/ObjectFieldMapTypeHandlerFactoryTest.java [hunk 1]
```diff
@@ -30,22 +35,81 @@ public class ObjectFieldMapTypeHandlerFactoryTest {
...
+    @Test
+    @DisplayName("Test implicit type handler loading for plain old java objects (pojo)")
+    public void testPojo() {
+        typeHandlerFactory.create(new TypeInfo<SingleTypeClass<Integer>>() { }, context);
+
+        // Verify that the Integer TypeHandler loading was called on the TypeHandlerLibrary
+        verify(typeHandlerLibrary, times(1)).getTypeHandler((Type) any());
+        verify(typeHandlerLibrary, times(1)).getTypeHandler(eq(TypeInfo.of(Integer.class).getType()));
+        verify(typeHandlerLibrary, never()).getTypeHandler((Class<Object>) any());
+        verify(typeHandlerLibrary, never()).getTypeHandler((TypeInfo<Object>) any());
```
Comment (skaldarnar) [inline subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/coreTypes/factories/ObjectFieldMapTypeHandlerFactoryTest.java:67]: I'd like to explain what we are doing here at least once.

Also, I'd move the specific test up (test happy path) and then make the "negative tests" afterwards.

```suggestion
	  // verify that the expected typehandler was created
        verify(typeHandlerLibrary, times(1)).getTypeHandler(eq(TypeInfo.of(Integer.class).getType()));
        
        // verify that creation was triggered only for the expected typehandler, but not for others
        // 'getTypeHandler'  can be called with either Type, Class, or TypeInfo, and we only expect one call with Type
        verify(typeHandlerLibrary, times(1)).getTypeHandler((Type) any());
        verify(typeHandlerLibrary, never()).getTypeHandler((Class<Object>) any());
        verify(typeHandlerLibrary, never()).getTypeHandler((TypeInfo<Object>) any());
```

**Review by skaldarnar — APPROVED** [review]

**Comment by jdrueckert** [comment]
Failing test `ExampleTest.testClientConnection` is a known sporadic and unrelated to code changes in this PR.
Merging despite the failing test.
